File: back-end/src/service/certificate_service.py
from src.service import Service
from src.data.models.certificate import Certificate

class CertificateService(Service):

    def __init__(self):
        self.certificate = Certificate()

    def get_certificates(self):
        return self.certificate.get_all_certificates()

    def get_certificate(self, certificate_id):
        return self.certificate.get_certificate(certificate_id)

    def insert_certificate(self, data):
        return self.certificate.insert_certificate(data)

    def delete_certificate(self, certificate_id):
        return self.certificate.delete_certificate(certificate_id)

    def update_certificate(self, data):
        return self.certificate.update_certificate(data)

    def get_my_certificates(self, core_app_context):
        return self.certificate.get_my_certificates(core_app_context)
    
    def get_event_certificates(self, event_id):
        return self.certificate.get_event_certificates(event_id)

File: front-end/src/app/core/services/certificate.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { API_ENDPOINT } from '../config';
import { Certificate } from './../models/certificate.model';

@Injectable({
  providedIn: 'root'
})
export class CertificateService {

  constructor(
    private httpClient: HttpClient
  ) { }

  getCertificates(): Observable<any> {
    return this.httpClient.get(API_ENDPOINT + '/certificate/all').pipe(
      map(res => res)
    )
  }

  getMyCertificates(): Observable<any> {
    return this.httpClient.get(API_ENDPOINT + '/certificate/my-certificates').pipe(
      map(res => res)
    )
  }
  // /api/v1/certificate/<certificate_id>/validate-certificate
  getById(id: string): Observable<any> {
    return this.httpClient.get(API_ENDPOINT + '/certificate/' + id).pipe(
      map(res => res)
    );
  }

  getCertificate(id: string): Observable<any> {
    return this.httpClient.get(API_ENDPOINT + '/certificate/' + id + '/validate-certificate').pipe(
      map(res => res)
    );
  }

  createCertificate(certificate: Certificate): Observable<any> {
    return this.httpClient.post(API_ENDPOINT + '/certificate/insert', certificate).pipe(
      map(res => res)
    )
  }

  updateCertificate(certificate: Certificate): Observable<any> {
    return this.httpClient.put(API_ENDPOINT + '/certificate/', certificate).pipe(
      map(res => res)
    )
  }

  deleteCertificate(id: string): Observable<any> {
    return this.httpClient.delete(API_ENDPOINT + '/certificate/' + id).pipe(
      map(res => res)
    )
  }  
}
